.ado/pipelines/templates/stages-loadtest-azure.yaml

```diff
@@ -208,7 +229,13 @@ stages:
                 Invoke-WebRequest $resultUrl -OutFile "results/results.zip"
                 echo "*** Download logsUrl"
                 Invoke-WebRequest $logsUrl -OutFile "results/logs.zip"
+
+                if ($result.testResult -eq "FAILED") {
+                  throw "*** ERROR: Test $($result.testRunId) failed."
+                }
+
               }
+
               # todo - timeout?
             } while ($result.status -ne "DONE")
```

Comment:
are we sure there is no other end state except for DONE? I had that somewhere else already that it would stop with ERROR, or something else, then the pipeline would get stuck forever

Reply:
Not 100% sure and the API is still preview and the potential values for `status` are not further documented (see [here](https://docs.microsoft.com/en-us/rest/api/loadtesting/dataplane/test-run/get-test-run?tabs=HTTP)).  But it was properly tested. The test result itself is not in `$result.status` its in `$result.testResult`.

Reply:
ok, I'm fine to go with this. Otherwise we would just need to implement that TODO-timeout I guess